doubleml/utils/_estimation.py

```diff
@@ -341,3 +341,48 @@ def _set_external_predictions(external_predictions, learners, treatment, i_rep):
         else:
             ext_prediction_dict[learner] = None
     return ext_prediction_dict
+
+
+def _solve_quadratic_inequation(a: float, b: float, c: float):
+    """
+    Solves the quadratic inequation a*x^2 + b*x + c <= 0 and returns the intervals.
+
+    Parameters
+    ----------
+    a : float
+        Coefficient of x^2.
+    b : float
+        Coefficient of x.
+    c : float
+        Constant term.
+
+    Returns
+    -------
+    List[Tuple[float, float]]
+        A list of intervals where the inequation holds.
+    """
+    determinant = b**2 - 4 * a * c
+
+    if determinant > 0:
+        root2 = (-b + np.sqrt(determinant)) / (2 * a)
+        root1 = (-b - np.sqrt(determinant)) / (2 * a)
+        if a > 0:  # happy quadratic (parabola opens upwards)
+            return [(root1, root2)]
+        else:  # sad quadratic (parabola opens downwards)
+            # TODO: eze, check this is correct, I think original implentation is wrong
+            # return [(-np.inf, root1), (root2, np.inf)]
+            # if a < 0 and b = 0, for instance, root1 > root2 (x^2 + 4 < 0)
+            return [(-np.inf, min(root1, root2)), (max(root1, root2), np.inf)]
```

Comment:
You are right, I got the order of the roots wrong

Reply:
Remove the comments though

Reply:
just wondering if the min and max are a bit dirty, shouldn't if just be (-inf, root2), (root1, inf)? If I understand correctly with a < 0 -> root2 < root1

Reply:
Yep, agree that's better